err_stash.py

```diff
@@ -168,42 +330,74 @@ def ensure_pull_requests_target_same_branch(plans, from_branch_display_id):
+                    slug=plan.slug,
+                    id=pr.number if isinstance(pr, PullRequest) else pr['id'],
+                    url=get_self_url(pr),
+                    to_ref=pr.base.ref if isinstance(pr, PullRequest) else pr['toRef']['id'])
+                )
 
         error_lines.append('Fix those PRs and try again. ')
         error_lines.append('Alternately you can pass `--force` to force the merge with different targets!')
         raise CheckError(error_lines)
 
 
-def make_pr_link(api, project, slug, from_branch, to_branch):
+def make_pr_link(url, project, slug, from_branch, to_branch):
     """Generates a URL that can be used to create a PR"""
     from urllib.parse import urlencode
-    params = OrderedDict([('sourceBranch', from_branch), ('targetBranch', to_branch)])
-    base_url = '{url}/projects/{project}/repos/{slug}/compare/commits?'.format(url=api.url, project=project,
-                                                                               slug=slug)
-    return base_url + urlencode(params)
+    if 'github' in url:
+        result = '{url}/{organization}/{repo_name}/compare/{from_branch}'.format(
+            url=url, organization=project, repo_name=slug, from_branch=from_branch)
+    else:
+        base_url = '{url}/projects/{project}/repos/{slug}/compare/commits?'.format(
+            url=url, project=project, slug=slug)
+        result = base_url + urlencode(
+            OrderedDict([('sourceBranch', from_branch), ('targetBranch', to_branch)]))
+
+    return result
 
 
-def get_commits_about_to_be_merged_by_pull_requests(api, plans, from_branch):
+def get_commits_about_to_be_merged_by_pull_requests(stash_api, github_api, plans, from_branch):
     """Returns a summary of the commits in each PR that will be merged"""
     error_lines = []
     result = []
     default_branch = next(plan.to_branch for plan in plans if plan.to_branch)
+
+    def default_branch_exists(plan):
+        branch_name = default_branch.replace("refs/heads/", "")
+
+        if plan.comes_from_github:
+            branches = list(github_api.fetch_branches(
+                organization=plan.project, repo_name=plan.slug, branch_name=branch_name))
+        else:
+            branches = list(stash_api.fetch_branches(plan.project, plan.slug, branch_name))
+
+        return len(branches) > 0
+
     for plan in plans:
         if plan.to_branch:
             to_branch = plan.to_branch
-        elif list(api.fetch_branches(plan.project, plan.slug, default_branch.replace("refs/heads/", ""))):
+        elif default_branch_exists(plan):
             to_branch = default_branch
         else:
             to_branch = "refs/heads/master"
-        try:
-            commits = list(api.fetch_repo_commits(plan.project, plan.slug, from_branch, to_branch))
-        except stashy.errors.NotFoundException:
-            commits = []
+
+        if plan.comes_from_github:
+            commits = list(github_api.fetch_repo_commits(
```

Comment:
Stash's handling below is around a try/except, this is not necessary here?

Reply:
PyGithub returns an empty list in case there are no commits 
`repo.compare(to_branch, to_branch).commits == [] #True`
so, unless there's another reason to not find commits, I think it's safe.